core/plugins/org.polarsys.capella.core.data.core.properties/src/org/polarsys/capella/core/data/core/properties/fields/TitleBlockBasicElementGroup.java

```diff
@@ -0,0 +1,175 @@
+    this(parent, widgetFactory, true, true);
+  }
+
+  /**
+   * @param parent
+   * @param widgetFactory
+   * @param hasNameField
+   * @param hasContentField
+   */
+  public TitleBlockBasicElementGroup(Composite parent, TabbedPropertySheetWidgetFactory widgetFactory,
+      boolean hasNameField, boolean hasContentField) {
+    super(widgetFactory);
+
+    Group textGroup = widgetFactory.createGroup(parent, ICommonConstants.EMPTY_STRING);
+    textGroup.setLayout(new GridLayout(2, false));
+    GridData gd = new GridData(GridData.FILL_HORIZONTAL);
+    gd.horizontalSpan = 2;
+    textGroup.setLayoutData(gd);
+
+    // Name
+    if (hasNameField) {
+      nameTextField = createTextField(textGroup, Messages.getString("NamedElement.NameLabel"));
+    }
+    // Content
+    if (hasContentField) {
+      contentTextField = createTextField(textGroup, Messages.getString("NamedElement.ContentLabel"));
+    }
+  }
+
+  /**
+   * @param textGroup
+   * @param textLabel
+   */
+  private Text createTextField(Group textGroup, String textLabel) {
+    widgetFactory.createCLabel(textGroup, textLabel);
+
+    Text textField = widgetFactory.createText(textGroup, ICommonConstants.EMPTY_STRING);
+    textField.addFocusListener(this);
+    textField.addKeyListener(this);
+    textField.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+
+    return textField;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public void loadData(EObject semanticElement, String name, String content) {
+    super.loadData(semanticElement, null);
+
+    if (null != semanticElement) {
+      if (null != nameTextField)
+        setTextValue(nameTextField, name);
+      if (null != contentTextField)
+        setTextValue(contentTextField, content);
+    }
+  }
+
+  protected void setTextValue(Text text, String value) {
+    text.setText(value);
+  }
+
+  /**
+   * @param textField
+   *          text field to be filled
+   */
+  @Override
+  protected void fillTextField(Text textField) {
+    if (textField.equals(nameTextField)) {
+      setNameValue(semanticElement, nameTextField.getText());
+    } else if (textField.equals(contentTextField)) {
+      setContentValue(semanticElement, contentTextField.getText());
+    }
+  }
+
+  private void setNameValue(EObject object, final Object value) {
+    AbstractReadWriteCommand command = new AbstractReadWriteCommand() {
+      @Override
+      public void run() {
+        ((DAnnotation) object).getDetails().put("Name:", value.toString());
+      }
+    };
+    executeCommand(command);
+  }
+
+  private void setContentValue(EObject object, final Object value) {
+    AbstractReadWriteCommand command = new AbstractReadWriteCommand() {
+      @Override
+      public void run() {
+        ((DAnnotation) object).getDetails().put("Content:", value.toString());
+      }
+    };
+    executeCommand(command);
+  }
+
+  /**
+   *
+   */
+  public void clearNameField() {

```

Comment:
What is this method supposed to do? There's no feature "name" associated to a DAnnotation.

Reply:
This is used for the name of the cell block.

Reply:
Sorry I was not clear.
setDataValue(semanticElement, ModellingcorePackage.eINSTANCE.getAbstractNamedElement_Name(), ""); is not necessary here. semanticElement is a DAnnotation which does not take part in the ModellingcorePackage and does not have a name.

Reply:
Removed.